server/src/server.js

```diff
@@ -659,6 +659,23 @@ app.get("/api/check-login-cookie", function(req, res) {
   }
 });
 
+/** This endpoint is used by wantsauth.js to check and retrieve
+    accountId cookie value */
+app.get("/api/get-accountId-cookie", function(req, res) {
+  if (!req.deviceId) {
+    sendRavenMessage(req, "Attempt to get accountid without login");
+    simpleResponse(res, "Not logged in", 401);
+    return;
+  }
+  const cookies = new Cookies(req, res, {keys: dbschema.getKeygrip()});
+  const accountId = cookies.get("accountid", {signed: true});
+  if (accountId) {
+    simpleResponse(res, accountId, 200);
+  } else {
+    simpleResponse(res, null, 401);
+  }
+});
+
```

Comment:
This endpoint isn't necessary.  `api/login` already look up the account id, so let's make use of that.  `auth.js` could get that for `sitehelper.js`, which then hands it to `wantsauth.js` on "login-successful".